### ISSUE TITLE
replace adaptNavigationTheme with getAdaptedTheme

### ISSUE THREAD

**Issue by davisk4rpi** [callstack/react-native-paper#3476]
**Is your feature request related to a problem? Please describe.**
`adaptNavigationTheme` does not work with custom Material themes and the input/output interface feels clunky.

**Describe the solution you'd like**
within  the function body of `adaptNavigationTheme`, there is a `getAdaptedTheme` function.  If this were exposed instead of `adaptNavigationTheme` we'd have a more flexible tool.

ex implementation:
```
  export function getAdaptedNavigationTheme(navigationTheme: NavigationTheme, theme: MD3Theme): NavigationTheme  {
    return {
      ...navigationTheme,
      colors: {
        ...navigationTheme.colors,
        primary: theme.colors.primary,
        background: theme.colors.background,
        card: theme.colors.elevation.level2,
        text: theme.colors.onSurface,
        border: theme.colors.outline,
        notification: theme.colors.error
      }
    };
  };
 ```

**Describe alternatives you've considered**
add an optional param to `adaptNavigationTheme` where you can provide custom light and dark MD3Themes

ex:
```
type OptionalMD3Theme = {
  light: MD3LightTheme;
  dark: MD3DarkTheme;
}
```

**Additional context**
version: `5.0.0-rc.10`

**Comment by lukewalczak** [comment]
> adaptNavigationTheme does not work with custom Material themes and the input/output interface feels clunky.

Could you please explain what doesn't work? Maybe you can present on the snack the current result vs your expectation?

**Comment by davisk4rpi** [comment]
@lukewalczak 

Im not sure that a snack is helpful here. I am not saying anything is broken, the function works as I would expect it to. Maybe rewording this as a question will clarify what I am requesting. 

In the example below, can I use `adaptNavigationTheme` to create a a navigation theme that has `colors.background === '#0FF'` ?

```
import {
  MD3LightTheme,
  MD3Theme,
  adaptNavigationTheme,
} from 'react-native-paper';

import {
  DefaultTheme as NavigationLightTheme,
} from '@react-navigation/native';


const customPaperTheme: MD3Theme = {
  ...MD3LightTheme,
  colors: {
    ...MD3LightTheme.colors,
    background: '#0FF',
  },
};

const { LightTheme } = adaptNavigationTheme(
  {light: NavigationLightTheme}
);
```

**Comment by lukewalczak** [comment]
Hey @davisk4rpi, thanks for the feedback, I see your point. I will try to refactor the utility to consider the presented case.

**Comment by davisk4rpi** [comment]
@lukewalczak Thank you, I'm happy to contribute if you would like any assistance.

**Comment by lukewalczak** [comment]
@davisk4rpi currently I have a plan for how to tackle that problem - probably I'll do that on Monday. 

I'm planning to create a hook based on the currently existing function `adaptNavigationTheme` which will be extended a bit to accept the user's theme, something like that:

`function adaptNavigationTheme(themes, userTheme)`

and then:

```
export function useAdaptNavigationTheme(themes) {
	const defaultTheme = useTheme() // user's theme passed to the Provider
	return adaptNavigationTheme(themes, defaultTheme)
```

If you have some time and can check if that approach solves the problem, please let me know ✌🏽

**Comment by davisk4rpi** [comment]
@lukewalczak Yes, this would solve my problem.  Thank you 😄

**Comment by lukewalczak** [comment]
Hey @davisk4rpi, I've raised the PR with the final approach which:
* extends `adaptNavigationThemes` by two new (optional) theme properties: `materialLight` and `materialDark`where you can pass you custom theme compliant with MD and Paper.
* existing properties `light` and `dark` were renamed to `reactNavigationLight` and `reactNavigationDark`
